# Mini Project 5-5 Explore Hypothesis Testing

## Introduction

You work for an environmental think tank called Repair Our Air (ROA). ROA is formulating policy recommendations to improve the air quality in America, using the Environmental Protection Agency's Air Quality Index (AQI) to guide their decision making. An AQI value close to 0 signals "little to no" public health concern, while higher values are associated with increased risk to public health. 

They've tasked you with leveraging AQI data to help them prioritize their strategy for improving air quality in America.

ROA is considering the following decisions. For each, construct a hypothesis test and an accompanying visualization, using your results of that test to make a recommendation:

1. ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.
2. With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?
3. A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?

**Notes:**
1. For your analysis, you'll default to a 5% level of significance.
2. Throughout the lab, for two-sample t-tests, use Welch's t-test (i.e., setting the `equal_var` parameter to `False` in `scipy.stats.ttest_ind()`). This will account for the possibly unequal variances between the two groups in the comparison.

## Step 1: Imports

To proceed with your analysis, import `pandas` and `numpy`. To conduct your hypothesis testing, import `stats` from `scipy`.

#### Import Packages

In [1]:
import pandas as pd
import numpy as np
from scipy import stats


You are also provided with a dataset with national Air Quality Index (AQI) measurements by state over time for this analysis. `Pandas` was used to import the file `c4_epa_air_quality.csv` as a dataframe named `aqi`. As shown in this cell, the dataset has been automatically loaded in for you. You do not need to download the .csv file, or provide more code, in order to access the dataset and proceed with this lab. Please continue with this activity by completing the following instructions.

**Note:** For purposes of your analysis, you can assume this data is randomly sampled from a larger population.

#### Load Dataset

In [2]:
df = pd.read_csv("c4_epa_air_quality.csv")


## Step 2: Data Exploration

### Before proceeding to your deliverables, explore your datasets.

Use the following space to surface descriptive statistics about your data. In particular, explore whether you believe the research questions you were given are readily answerable with this data.

In [3]:
# Use head() to show a sample of data
df.head()


,Unnamed: 0,date_local,state_name,county_name,city_name,local_site_name,parameter_name,units_of_measure,arithmetic_mean,aqi
0,0,2018-01-01,Arizona,Maricopa,Buckeye,BUCKEYE,Carbon monoxide,Parts per million,0.473684,7
1,1,2018-01-01,Ohio,Belmont,Shadyside,Shadyside,Carbon monoxide,Parts per million,0.263158,5
2,2,2018-01-01,Wyoming,Teton,Not in a city,Yellowstone National Park - Old Faithful Snow ...,Carbon monoxide,Parts per million,0.111111,2
3,3,2018-01-01,Pennsylvania,Philadelphia,Philadelphia,North East Waste (NEW),Carbon monoxide,Parts per million,0.300000,3
4,4,2018-01-01,Iowa,Polk,Des Moines,CARPENTER,Carbon monoxide,Parts per million,0.215789,3


In [4]:
df.columns


Index(['Unnamed: 0', 'date_local', 'state_name', 'county_name', 'city_name',
       'local_site_name', 'parameter_name', 'units_of_measure',
       'arithmetic_mean', 'aqi'],
      dtype='object')

In [5]:
df['aqi'].describe()


count    260.000000
mean       6.757692
std        7.061707
min        0.000000
25%        2.000000
50%        5.000000
75%        9.000000
max       50.000000
Name: aqi, dtype: float64

In [6]:
df['state_name'].value_counts()


state_name
California              66
Arizona                 14
Ohio                    12
Florida                 12
Texas                   10
New York                10
Pennsylvania            10
Michigan                 9
Colorado                 9
Minnesota                7
New Jersey               6
Indiana                  5
North Carolina           4
Massachusetts            4
Maryland                 4
Oklahoma                 4
Virginia                 4
Nevada                   4
Connecticut              4
Kentucky                 3
Missouri                 3
Wyoming                  3
Iowa                     3
Hawaii                   3
Utah                     3
Vermont                  3
Illinois                 3
New Hampshire            2
District Of Columbia     2
New Mexico               2
Montana                  2
Oregon                   2
Alaska                   2
Georgia                  2
Washington               2
Idaho                    2
Nebraska         

#### **Question 1: From the preceding data exploration, what do you recognize?**

A: From the data exploration, I observe that:

The dataset contains air quality index (AQI) values recorded across multiple U.S. states.

The aqi variable shows a range of values with some variation in spread (visible through .describe()).

Some states have more AQI observations than others, which could affect the balance and fairness of comparisons during hypothesis testing.

The variables available (like state_name, aqi, and date_local) seem sufficient to test hypotheses comparing AQI between states or across time.

## Step 3. Statistical Tests

Before you proceed, recall the following steps for conducting hypothesis testing:

1. Formulate the null hypothesis and the alternative hypothesis.<br>
2. Set the significance level.<br>
3. Determine the appropriate test procedure.<br>
4. Compute the p-value.<br>
5. Draw your conclusion.

### Hypothesis 1: ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.

Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [7]:
# Create dataframes for each sample being compared in your test
la_data = df[(df['state_name'] == 'California') & (df['county_name'] == 'Los Angeles')]
rest_ca_data = df[(df['state_name'] == 'California') & (df['county_name'] != 'Los Angeles')]


In [8]:
# Check head
print("LA County Sample:")
print(la_data['aqi'].head())

print("\nRest of California Sample:")
print(rest_ca_data['aqi'].head())


LA County Sample:
33      7
42      7
61     16
76     17
109    17
Name: aqi, dtype: int64

Rest of California Sample:
16    11
18    16
26    13
27    15
34    15
Name: aqi, dtype: int64


#### Formulate your hypothesis:

**Formulate your null and alternative hypotheses:**

*   $H_0$: There is no difference in the mean AQI between Los Angeles County and the rest of California.
*   $H_A$: There is a difference in the mean AQI between Los Angeles County and the rest of California.


#### Set the significance level:

In [9]:

alpha = 0.05  # 5% significance level


#### Determine the appropriate test procedure:

Here, you are comparing the sample means between two independent samples. Therefore, you will utilize a **two-sample  𝑡-test**.

#### Compute the P-value

In [ ]:
# Compute your p-value here
from scipy.stats import ttest_ind

# Run Welch’s t-test
t_stat, p_val = ttest_ind(
    la_data['aqi'], 
    rest_ca_data['aqi'], 
    equal_var=False
)

print("T-statistic:", t_stat)



T-statistic: 2.1107010796372014
P-value: 0.049839056842410995


#### **Question 2. What is your P-value for hypothesis 1, and what does this indicate for your null hypothesis?**

In [12]:
# Extracting pvalue and make the test
print("P-value:", p_val)

P-value: 0.049839056842410995


A: This means that there is a statistically significant difference in the mean AQI between Los Angeles County and the rest of California. In other words, Los Angeles has a distinct air quality pattern compared to the rest of the state.

### Hypothesis 2: With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?

Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [13]:
# Create dataframes for each sample being compared in your test
ny_data = df[df['state_name'] == 'New York']['aqi']
oh_data = df[df['state_name'] == 'Ohio']['aqi']


In [14]:
# Check head
# Preview the first few rows
print("New York AQI sample:")
print(ny_data.head())

print("\nOhio AQI sample:")
print(oh_data.head())


New York AQI sample:
90     3
113    3
124    2
167    2
173    3
Name: aqi, dtype: int64

Ohio AQI sample:
1     5
12    3
22    6
51    3
59    3
Name: aqi, dtype: int64


**Formulate your null and alternative hypotheses:**

*   $H_0$: The mean AQI of New York is greater than or equal to that of Ohio.
*   $H_A$: The mean AQI of New York is **below** that of Ohio.


#### Significance Level (remains at 5%)

#### Determine the appropriate test procedure:

Here, you are comparing the sample means between two independent samples in one direction. Therefore, you will utilize a **two-sample  𝑡-test**.

#### Compute the P-value

In [15]:
from scipy import stats

# Subset data
ny_data = df[df['state_name'] == 'New York']['aqi']
oh_data = df[df['state_name'] == 'Ohio']['aqi']

# Two-sample t-test with unequal variance (Welch’s t-test)
t_stat, p_two_sided = stats.ttest_ind(ny_data, oh_data, equal_var=False)

# One-sided p-value for Hₐ: New York AQI < Ohio AQI
p_one_sided = p_two_sided / 2 if t_stat < 0 else 1 - (p_two_sided / 2)

# Output
print("T-statistic:", t_stat)
print("One-sided P-value:", p_one_sided)


T-statistic: -2.025951038880333
One-sided P-value: 0.030446502691934683


#### **Question 3. What is your P-value for hypothesis 2, and what does this indicate for your null hypothesis?**

In [16]:
# Your code here.
print("One-sided P-value:", p_one_sided)

One-sided P-value: 0.030446502691934683


A: The one-sided P-value is 0.0304.
Since this is less than the significance level of 0.05, we reject the null hypothesis.
This indicates that the mean AQI in New York is significantly lower than that in Ohio.

###  Hypothesis 3: A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?

Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [17]:
# Create dataframes for each sample being compared in your test
# Subset the data for Michigan
mi_data = df[df['state_name'] == 'Michigan']['aqi']


**Formulate your null and alternative hypotheses here:**

*   $H_0$: The mean AQI of Michigan is less than or equal to 10.
*   $H_A$: The mean AQI of Michigan is greater than 10.


#### Significance Level (remains at 5%)

#### Determine the appropriate test procedure:

Here, you are comparing one sample mean relative to a particular value in one direction. Therefore, you will utilize a **one-sample  𝑡-test**. 

#### Compute the P-value

In [18]:
# Compute your p-value here
from scipy import stats

# Null hypothesis value
mu = 10

# Perform one-sample, one-sided t-test (Michigan mean > 10)
t_stat, p_two_sided = stats.ttest_1samp(mi_data, popmean=mu)

# Convert to one-sided p-value
p_one_sided = p_two_sided / 2 if t_stat > 0 else 1 - (p_two_sided / 2)

# Output
print("T-statistic:", t_stat)
print("One-sided P-value:", p_one_sided)


T-statistic: -1.7395913343286131
One-sided P-value: 0.9399405193140109


#### **Question 4. What is your P-value for hypothesis 3, and what does this indicate for your null hypothesis?**

A: The one-sided P-value for Hypothesis 3 is 0.9399.
Since this is much greater than 0.05, we fail to reject the null hypothesis.
This means there is not enough evidence to conclude that Michigan's mean AQI is greater than 10.

## Step 4. Results and Evaluation

Now that you've completed your statistical tests, you can consider your hypotheses and the results you gathered.

#### **Question 5. Did your results show that the AQI in Los Angeles County was statistically different from the rest of California?**

A: Yes. The p-value from the test was 0.0489, which is less than 0.05.
So we reject the null hypothesis, indicating that the AQI in Los Angeles County is statistically different from the rest of California.

#### **Question 6. Did New York or Ohio have a lower AQI?**

A: The one-sided P-value was 0.0304, which is less than 0.05, meaning we reject the null hypothesis.
Therefore, New York had a significantly lower AQI than Ohio.

#### **Question 7: Will Michigan be affected by the new policy impacting states with a mean AQI of 10 or greater?**



A: No. The test resulted in a high P-value (0.9399) and a negative t-statistic, meaning we do not reject the null hypothesis.
This implies that Michigan's mean AQI is not significantly greater than 10, so it is not likely to be affected by the policy.

# Conclusion

**What are key takeaways from this project?**

A: Hypothesis testing is a powerful tool for determining statistically significant differences between groups.

Los Angeles County's AQI differs significantly from the rest of California.

New York has a statistically lower AQI than Ohio.

Michigan is unlikely to be impacted by AQI-related policies targeting high-pollution states.

**What would you consider presenting to your manager as part of your findings?**

A: Los Angeles County should be monitored separately for AQI regulations.

New York presents a cleaner air profile compared to Ohio—potentially a better site for expansion.

Michigan’s AQI currently does not justify concern under stricter pollution policy thresholds.


**What would you convey to external readers?**

A: The analysis confirms measurable air quality differences across regions. It provides a data-driven lens for prioritizing regulatory efforts and selecting locations for expansion with minimal environmental impact.
